core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -257,6 +284,17 @@ public ClientRequestContext newDerivedContext(Request request, Endpoint endpoint
         return new DefaultClientRequestContext(this, request, endpoint);
     }
 
+    @Override
+    protected void validateHeaders(RequestHeaders headers) {
+        // Do not validate/filter if the context is not fully initialized yet,
+        // because init() will trigger this method again via updateEndpoint().
+        if (eventLoop == null) {
```

Comment:
https://github.com/line/armeria/pull/2092/files#diff-d437ddb5e0b4572dba776949bacf9becR127
`eventLoop` can be passed as one of the constructor arguments. So I think another flag for that. 😉

Reply:
Doh! Fixed by adding `initialized` flag.